src/parser/mod.rs

```diff
@@ -6251,10 +6265,30 @@ impl<'a> Parser<'a> {
                     Keyword::REPLACE,
                 ])?,
             )))
+        } else if self.parse_keywords(&[Keyword::TAG])
```

Comment:
To avoid prematurely consuming the with and then later potentially rewinding to prev token, we can probably do something like

```rust
else if (self.parse_keyword([TAG) || self.parse_keywords([WITH, TAG])) && dialect_of()
```
Also can we switch the dialect_of for a dialect method similar to the others? Related question: is the tags support here so that we can also cover generic dialect? (if so and we didn't have support for the generic dialect previously, then I'm thinking we can just keep it limited to snowflake and move the code over to the dialect)

Reply:
A parsing of `TAG` column options moved to a snowflake dialect implementation.